pkg/controller/cronjob/utils.go

```diff
@@ -160,13 +142,18 @@ func mostRecentScheduleTime(cj *batchv1.CronJob, now time.Time, schedule cron.Sc
 // realistic cases should be around 100s, the job will still be executed without missing
 // the schedule.
 func nextScheduleTimeDuration(cj *batchv1.CronJob, now time.Time, schedule cron.Schedule) *time.Duration {
-	earliestTime, mostRecentTime, _, err := mostRecentScheduleTime(cj, now, schedule, false)
+	earliestTime, mostRecentTime, numberOfMissedSchedules, err := mostRecentScheduleTime(cj, now, schedule, false)
 	if err != nil {
 		// we still have to requeue at some point, so aim for the next scheduling slot from now
 		mostRecentTime = &now
 	} else if mostRecentTime == nil {
-		// no missed schedules since earliestTime
-		mostRecentTime = &earliestTime
+		if numberOfMissedSchedules > 0 {
+			// if there are missed schedules since earliestTime
+			mostRecentTime = &now
+		} else {
+			// no missed schedules since earliestTime
+			mostRecentTime = &earliestTime
+		}
```

Comment:
This above condition is the critical piece we want back, but I'm still not inclined to leak that misleading information about number of missed schedules out of `mostRecentScheduleTime`. So I'm proposing alternative approach in https://github.com/kubernetes/kubernetes/pull/121327

Reply:
/lgtm